include/dlaf/multiplication/general/impl.h

```diff
@@ -19,13 +19,50 @@
 #include <dlaf/communication/communicator_grid.h>
 #include <dlaf/matrix/distribution.h>
 #include <dlaf/matrix/index.h>
+#include <dlaf/matrix/matrix.h>
+#include <dlaf/matrix/matrix_ref.h>
 #include <dlaf/matrix/panel.h>
 #include <dlaf/multiplication/general/api.h>
 #include <dlaf/sender/when_all_lift.h>
+#include <dlaf/util_matrix.h>
 
 namespace dlaf::multiplication {
 namespace internal {
 
+template <Backend B, Device D, class T>
+void General<B, D, T>::callNN(const T alpha, MatrixRef<const T, D>& mat_a, MatrixRef<const T, D>& mat_b,
+                              const T beta, MatrixRef<T, D>& mat_c) {
+  namespace ex = pika::execution::experimental;
+
+  using matrix::multipliable_sizes;
+  DLAF_ASSERT_HEAVY(multipliable_sizes(mat_a.size(), mat_b.size(), mat_c.size(), blas::Op::NoTrans,
+                                       blas::Op::NoTrans),
+                    "Multiplication incompatible matrix sizes.", mat_a.size(), mat_b.size(),
```

Comment:
Is the string needed?
The error message already reports that `multipliable_sizes(mat_a.size(), mat_b.size(), mat_c.size(), blas::Op::NoTrans, blas::Op::NoTrans)` failed and it is self explanatory.